core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1085,17 +1084,14 @@ object GroupMetadataManager {
    * Generates the payload for offset commit message from given offset and metadata
    *
    * @param offsetAndMetadata consumer's current offset and metadata
-   * @param metadataVersion the api version
+   * @param maxVersion the naxinu -  // Serialize with the highest supported non-flexible version
+      // until a tagged field is introduced or the version is bumped. FIXME
    * @return payload for offset commit message
    */
-  def offsetCommitValue(offsetAndMetadata: OffsetAndMetadata,
-                        metadataVersion: MetadataVersion): Array[Byte] = {
+  def offsetCommitValue(offsetAndMetadata: OffsetAndMetadata, maxVersion: Short = 3): Array[Byte] = {
     val version =
-      if (metadataVersion.isLessThan(IBP_2_1_IV0) || offsetAndMetadata.expireTimestampMs.isPresent) 1.toShort
-      else if (metadataVersion.isLessThan(IBP_2_1_IV1)) 2.toShort
-      // Serialize with the highest supported non-flexible version
-      // until a tagged field is introduced or the version is bumped.
-      else 3.toShort
+      if (offsetAndMetadata.expireTimestampMs.isPresent) 1.toShort
+      else maxVersion
```

Comment:
Is there a need to pass it as an argument to the method or could we just hardcode it?

Reply:
This is needed for some tests that verify the ability to read messages encoded with the old format. I'm not sure if there's a way to make sure there are no messages written with the old format by the time the broker is upgraded to 4.0. Given the usual retention rules, it should be fine, but not sure if we can be sure. Thoughts?